libs/src/services/solanaWeb3Manager/rewardsAttester.js

```diff
@@ -476,6 +479,14 @@ class RewardsAttester {
   async _delay (waitTime) {
     return new Promise(resolve => setTimeout(resolve, waitTime))
   }
+
+  async _addRecentlyDisbursed (challenges) {
+    const ids = challenges.map(this._disbursementToKey)
+    this.recentlyDisbursedQueue.push(...ids)
+    if (this.recentlyDisbursedQueue.length > MAX_DISBURSED_CACHE_SIZE) {
+      this.recentlyDisbursedQueue.splice(0, this.recentlyDisbursedQueue.length - MAX_DISBURSED_CACHE_SIZE)
```

Comment:
Would it be possible / be an issue if the len of challenges was greater than MAX_DISBURSED_CACHE_SIZE - then some of the challenges would never make it into the recently disbursed queue

Reply:
yeah that would be an issue - don't think we'd ever go up to 100 challenges in parallel though (hitting 3 DPs and AAO with 100 concurrent requests each). I think safe to leave this as is for now